commit e12c532310de5d75947ee03fe90a53ad08fc4d27
Author: gwx <wx.gu@netxeon.com>
Date:   Mon Feb 25 09:22:13 2019 +0800

    delay boot animation

diff --git a/drivers/amlogic/media/osd/osd_fb.c b/drivers/amlogic/media/osd/osd_fb.c
index 9e950f7..0f97705 100644
--- a/drivers/amlogic/media/osd/osd_fb.c
+++ b/drivers/amlogic/media/osd/osd_fb.c
@@ -4316,7 +4316,7 @@ static int osd_probe(struct platform_device *pdev)
 		vout2_register_client(&osd_notifier_nb2);
 #endif
 	INIT_DELAYED_WORK(&osd_dwork, mem_free_work);
-	schedule_delayed_work(&osd_dwork, msecs_to_jiffies(60 * 1000));
+	schedule_delayed_work(&osd_dwork, msecs_to_jiffies(120 * 1000));
 
 	osd_log_info("osd probe OK\n");
 	return 0;
