diff --git a/drivers/amlogic/media/osd/osd_hw.c b/drivers/amlogic/media/osd/osd_hw.c
index 1afcc95..b45d00d 100644
--- a/drivers/amlogic/media/osd/osd_hw.c
+++ b/drivers/amlogic/media/osd/osd_hw.c
@@ -7535,8 +7535,8 @@ static int osd_setting_order(void)
 	osd_log_dbg2(MODULE_RENDER,
 		"enter osd_setting_order:cnt=%d,encp line=%d\n",
 		cnt, line2);
-	if (line2 < line1)
-		osd_log_info("osd line %d,%d\n", line1, line2);
+	//if (line2 < line1)
+		//osd_log_info("osd line %d,%d\n", line1, line2);
 	osd_wait_vsync_hw();
 	return 0;
 }
