commit e2d49a3e89d654e7abb0adab3c5a8d55e9401074
Author: Eric <zrb@trigtop.com>
Date:   Sun Sep 29 15:24:42 2019 +0800

    for led

diff --git a/drivers/amlogic/pm/lgcy_early_suspend.c b/drivers/amlogic/pm/lgcy_early_suspend.c
index 4cc4b09..cdee01d 100644
--- a/drivers/amlogic/pm/lgcy_early_suspend.c
+++ b/drivers/amlogic/pm/lgcy_early_suspend.c
@@ -91,7 +91,7 @@ static inline void early_suspend(void)
 		already_early_suspend = 1;
 	else
 		goto end_early_suspend;
-
+	set_led_value_output(0);
 	pr_info("early_suspend: call handlers\n");
 	list_for_each_entry(pos, &early_suspend_handlers, link)
 		if (pos->suspend != NULL) {
@@ -124,6 +124,7 @@ static inline void late_resume(void)
 			pr_info("late_resume: %pf\n", pos->resume);
 			pos->resume(pos);
 		}
+	set_led_value_output(1);
 	pr_info("late_resume: done\n");
 
 end_late_resume:
diff --git a/drivers/leds/leds-gpio.c b/drivers/leds/leds-gpio.c
index d400dca..a9d4d01 100644
--- a/drivers/leds/leds-gpio.c
+++ b/drivers/leds/leds-gpio.c
@@ -29,6 +29,7 @@ struct gpio_led_data {
 	gpio_blink_set_t platform_gpio_blink_set;
 };
 
+struct led_classdev *gu_led;
 static inline struct gpio_led_data *
 			cdev_to_gpio_led_data(struct led_classdev *led_cdev)
 {
@@ -138,9 +139,15 @@ static int create_gpio_led(const struct gpio_led *template,
 	ret = gpiod_direction_output(led_dat->gpiod, state);
 	if (ret < 0)
 		return ret;
-
+	gu_led = &led_dat->cdev ;
 	return devm_led_classdev_register(parent, &led_dat->cdev);
 }
+extern void set_led_value_output(int flag)
+{
+       gpio_led_set_blocking(gu_led , (enum led_brightness)flag);
+
+}
+
 
 struct gpio_leds_priv {
 	int num_leds;
diff --git a/include/linux/suspend.h b/include/linux/suspend.h
index 90d8569..9c083cc 100644
--- a/include/linux/suspend.h
+++ b/include/linux/suspend.h
@@ -266,7 +266,7 @@ static inline bool idle_should_freeze(void)
  * done. Not called for suspend to disk.
  */
 extern void arch_suspend_enable_irqs(void);
-
+extern void set_led_value_output(int flag);
 extern int pm_suspend(suspend_state_t state);
 #else /* !CONFIG_SUSPEND */
 #define suspend_valid_only_mem	NULL
