diff --git a/arch/arm/boot/dts/amlogic/g12b_a311d_w400_a.dts b/arch/arm/boot/dts/amlogic/g12b_a311d_w400_a.dts
index f65b86c..5e980f6 100644
--- a/arch/arm/boot/dts/amlogic/g12b_a311d_w400_a.dts
+++ b/arch/arm/boot/dts/amlogic/g12b_a311d_w400_a.dts
@@ -1349,6 +1349,10 @@
 &uart_A {
 	status = "okay";
 };
+&uart_AO_B{
+	status = "okay";
+ };
+
 
 &pcie_A {
 	reset-gpio = <&gpio GPIOX_7 GPIO_ACTIVE_HIGH>;
diff --git a/arch/arm/boot/dts/amlogic/mesong12b_a.dtsi b/arch/arm/boot/dts/amlogic/mesong12b_a.dtsi
index 69b03f6..88052c6 100644
--- a/arch/arm/boot/dts/amlogic/mesong12b_a.dtsi
+++ b/arch/arm/boot/dts/amlogic/mesong12b_a.dtsi
@@ -496,7 +496,7 @@
 
 	jtag {
 		compatible = "amlogic, jtag";
-		status = "okay";
+		status = "disable";
 		select = "disable"; /* disable/apao/apee */
 		pinctrl-names="jtag_apao_pins", "jtag_apee_pins";
 		pinctrl-0=<&jtag_apao_pins>;
@@ -788,7 +788,7 @@
 				compatible = "amlogic, meson-uart";
 				reg = <0x4000 0x18>;
 				interrupts = <0 197 1>;
-				status = "disabled";
+				status = "okay";
 				clocks = <&xtal>;
 				clock-names = "clk_uart";
 				fifosize = < 64 >;
@@ -1940,8 +1940,8 @@
 
 	ao_b_uart_pins:ao_b_uart {
 		mux {
-			groups = "uart_ao_tx_b_2",
-				"uart_ao_rx_b_3";
+			groups = "uart_ao_tx_b_8",
+				"uart_ao_rx_b_9";
 			function = "uart_ao_b";
 		};
 	};
