commit 147c9ee889d3b9c9e4ddd405fbb21d89725293f1
Author: gwx <wx.gu@netxeon.com>
Date:   Sat Mar 16 10:52:25 2019 +0800

    for usb3.0
    
    Conflicts:
    	arch/arm/boot/dts/amlogic/g12b_a311d_w400.dts

diff --git a/arch/arm/boot/dts/amlogic/g12b_a311d_w400.dts b/arch/arm/boot/dts/amlogic/g12b_a311d_w400.dts
index e2f1ff0..3a4a0b4 100644
--- a/arch/arm/boot/dts/amlogic/g12b_a311d_w400.dts
+++ b/arch/arm/boot/dts/amlogic/g12b_a311d_w400.dts
@@ -1438,7 +1438,7 @@
 &usb3_phy_v2 {
 	status = "okay";
 	portnum = <1>;
-	otg = <1>;
+	otg = <0>;
 	gpio-vbus-power = "GPIOH_6";
 	gpios = <&gpio GPIOH_6 GPIO_ACTIVE_HIGH>;
 };
@@ -1467,7 +1467,7 @@
 
 &pcie_A {
 	reset-gpio = <&gpio GPIOX_7 GPIO_ACTIVE_HIGH>;
-	status = "disable";
+	status = "dissble";
 };
 
 &saradc {
